src/vault.py

```diff
@@ -47,6 +48,15 @@ def is_sealed(self) -> bool:
         """Returns whether Vault is sealed."""
         return self._client.sys.is_sealed()
 
+    def wait_for_unseal(self) -> None:
+        """Wait for Vault to be unsealed."""
```

Comment:
This waits for Vault to be unsealed, up to a timeout. If it takes longer than that, we proceed anyway.

I'm not sure if that is intentional, but the docstring should probably make that clear. Perhaps we want to throw an exception if the Vault is still sealed after the timeout? Or log a warning? I'm not sure of the wider context and what would be more appropriate.

Reply:
Good point.
I think throwing an exception makes sense, as continuing with a sealed vault and then trying to create the audit device would throw an exception too.

Done.

Reply:
Do you think it would be a better approach if we call `vault.unseal` in `wait_for_unseal`? or should we keep it as it is now?

Reply:
IMO,  If a single method includes all the logic for unsealing including the wait for timeout, it would be better.